core/services/workflows/v2/engine.go

```diff
@@ -167,7 +168,7 @@ func (e *Engine) runTriggerSubscriptionPhase(ctx context.Context) error {
 		Request:         &wasmpb.ExecuteRequest_Subscribe{},
 		MaxResponseSize: uint64(e.cfg.LocalLimits.ModuleExecuteMaxResponseSizeBytes),
 		// no Config needed
-	}, &DisallowedExecutionHelper{})
+	}, NewVaultOnlyExecutionHelper(e, e.cfg.WorkflowOwner, workflowkey.Key{}))
```

Comment:
hmmmm I didn't realize that blocking Secrets call can happen here. That might affect the timeout values that we set. Also it means we should probably make the subscription phase retryable if we fail to fetch secrets... it will add some complexity here but it looks like it should be done (not necessarily in this PR).